test/interfaces/background_field_interface.jl

```diff
@@ -0,0 +1,231 @@
+            atol=ATOL,
+            rtol=RTOL,
+        )
+        @test isapprox(
+            envelope(test_field, rightendpoint(RND_DOMAIN) + eps()),
+            zero(Float64),
+            atol=ATOL,
+            rtol=RTOL,
+        )
+    end
+    @testset "compute vector" begin
+        rnd_phis = rand(RNG, RND_DOMAIN, 2)
+        push!(rnd_phis, leftendpoint(RND_DOMAIN))
+        push!(rnd_phis, leftendpoint(RND_DOMAIN) - eps())
+        push!(rnd_phis, rightendpoint(RND_DOMAIN))
+        push!(rnd_phis, rightendpoint(RND_DOMAIN) + eps())
+
+        test_envelope_values = envelope(test_field, rnd_phis)
+
+        groundtruth_envelope_values = Vector{Float64}(undef, length(rnd_phis))
+        for (idx, phi) in enumerate(rnd_phis)
+            if phi in RND_DOMAIN
+                groundtruth_envelope_values[idx] = _groundtruth_envelope(phi)
+            else
+                groundtruth_envelope_values[idx] = zero(phi)
+            end
+        end
+
+        @test isapprox(
+            test_envelope_values, groundtruth_envelope_values, atol=ATOL, rtol=RTOL
+        )
+    end
+end
+
+@testset "pulse amplitude" begin
+    @testset "$pol" for pol in (QEDfields.PolX(), QEDfields.PolY())
+        test_field = TestBGfield()
+
+        @testset "compute single" begin
+            rnd_phi = rand(RNG, RND_DOMAIN)
+            @test isapprox(
+                amplitude(test_field, pol, rnd_phi),
+                _groundtruth_amplitude(pol, rnd_phi),
+                atol=ATOL,
+                rtol=RTOL,
+            )
+            @test isapprox(
+                amplitude(test_field, pol, leftendpoint(RND_DOMAIN)),
+                _groundtruth_amplitude(pol, leftendpoint(RND_DOMAIN)),
+                atol=ATOL,
+                rtol=RTOL,
+            )
+            @test isapprox(
+                amplitude(test_field, pol, leftendpoint(RND_DOMAIN) - eps()),
+                zero(Float64),
+                atol=ATOL,
+                rtol=RTOL,
+            )
+            @test isapprox(
+                amplitude(test_field, pol, rightendpoint(RND_DOMAIN)),
+                _groundtruth_amplitude(pol, rightendpoint(RND_DOMAIN)),
+                atol=ATOL,
+                rtol=RTOL,
+            )
+            @test isapprox(
+                amplitude(test_field, pol, rightendpoint(RND_DOMAIN) + eps()),
+                zero(Float64),
+                atol=ATOL,
+                rtol=RTOL,
+            )
+        end
+
+        @testset "compute vector" begin
+            rnd_phis = rand(RNG, RND_DOMAIN, 2)
+            push!(rnd_phis, leftendpoint(RND_DOMAIN))
+            push!(rnd_phis, leftendpoint(RND_DOMAIN) - eps())
+            push!(rnd_phis, rightendpoint(RND_DOMAIN))
+            push!(rnd_phis, rightendpoint(RND_DOMAIN) + eps())
+
+            test_amplitude_values = amplitude(test_field, pol, rnd_phis)
+
+            groundtruth_amplitude_values = Vector{Float64}(undef, length(rnd_phis))
```

Comment:
Same as above. (use of uninitialized list)

Reply:
Same comment as above, but this time, the list is initialized by `-2.0` because the values of an amplitude vary between `-1` and `1`. Changes made in 73f7bfc